### PR TITLE
Feature/5437 get funding source

### PR DESCRIPTION
**Pull request by sam-roberts-bjss**
Story: [AB#5437](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/5437)
Task: [AB#8656](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8656)

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs [hunk 1]
```diff
@@ -18,17 +16,86 @@
...
         }
 
+        [Test]
+        public async Task GetAsync_OrderDoesNotExist_ReturnsNotFound()
+        {
+            const string orderId = "DOESNOTEXIST";
+            var context = FundingSourceControllerTestContext.Setup();
+            context.Order = null;
+
+            using var controller = context.Controller;
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs:38]: Do you have to extract out controller, why don't you just use it in the next line

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs [hunk 2]
```diff
@@ -18,17 +16,86 @@
...
+        {
+            const string orderId = "C0000014-01";
+            var context = FundingSourceControllerTestContext.Setup();
+            context.PrimaryOrganisationId = Guid.NewGuid();
+            context.Order = OrderBuilder
+                .Create()
+                .WithOrganisationId(Guid.NewGuid())
+                .Build();
+
+            using var controller = context.Controller;
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs:56]: Same here

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs [hunk 3]
```diff
@@ -18,17 +16,86 @@
...
+            var actual = await controller.GetAsync(orderId);
+
+            actual.Result.Should().BeOfType<ForbidResult>();
+        }
+
+        [Test]
+        public async Task GetAsync_OrderExists_FundingSourceDetailsReturned()
+        {
+            const string orderId = "C0000014-01";
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs:66]: Can you take away this new line, so it matches the pattern of the other tests you have done

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs [hunk 4]
```diff
@@ -18,17 +16,86 @@
...
+            var context = FundingSourceControllerTestContext.Setup();
+            context.Order = OrderBuilder
+                .Create()
+                .WithOrganisationId(context.PrimaryOrganisationId)
+                .WithFundingSourceOnlyGms(true)
+                .Build();
+
+            using var controller = context.Controller;
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs:74]: *here too

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs [hunk 5]
```diff
@@ -18,17 +16,86 @@
...
+        }
+
+        [Test]
+        public async Task GetAsync_GetOrderByIdAsync_CalledOnce()
+        {
+            const string orderId = "C0000014-01";
+
+            var context = FundingSourceControllerTestContext.Setup();
+
+            using var controller = context.Controller;
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs:92]: Here too

File: tests/NHSD.BuyingCatalogue.Ordering.Common.UnitTests/Builders/OrderBuilder.cs [hunk 6]
```diff
@@ -26,7 +26,7 @@ public sealed class OrderBuilder
         private bool _catalogueSolutionsViewed;
         private bool _additionalServicesViewed;
         private bool _associatedServicesViewed;
-        private bool? _fundingSourceOnlyGMS;
+        private bool? _fundingSourceOnlyGms;
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Common.UnitTests/Builders/OrderBuilder.cs:29]: Just curious, why did you change the name for this? Is it due to  resharper?

Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Common.UnitTests/Builders/OrderBuilder.cs:29]: Yeah Resharper complains about the capitals. I thought we'd said ages ago in one of the coding standards meetings we wouldn't capitalise acronyms e.g. API

Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Common.UnitTests/Builders/OrderBuilder.cs:29]: Ahhh, my bad. My memory isn't the best 😆

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs (hunk 1 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/FundingSourceControllerTests.cs:38]: I think it was a pattern used elsewhere, doesn't have to but will lose the `using` keyword. I will change them.

File: tests/NHSD.BuyingCatalogue.Ordering.Common.UnitTests/Builders/OrderBuilder.cs (hunk 6 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Common.UnitTests/Builders/OrderBuilder.cs:29]: 😂 That one only stuck with me because I introduced API somewhere & got shot down.
